src/utilities/validateConnectionArgs.ts

```diff
@@ -0,0 +1,19 @@
+import type { UserPostsArgs } from "../types/generatedGraphQLTypes";
```

Comment:
This is hard to make sense of. I'll just give you the code logic, here:-
```typescript
import { GraphQLError } from "graphql";

export function isNotNullish<T>(value: T | undefined | null): value is T {
  return value !== undefined && value !== null;
}

export type RelayConnectionArguments = {
  after?: string | null;
  before?: string | null;
  first?: number | null;
  last?: number | null;
};

export type ParsedRelayConnectionArguments = {
  cursor: string | null;
  limit: number;
  direction: "FORWARD" | "BACKWARD";
};

// some global or database record based hard limit for the number of records
// that can be returned in one connection query for the corresponding record
// replace this which an environment variable preferably
const env = {
  LIMIT: 0,
};

export function parseConnectionArguments(
  args: RelayConnectionArguments,
): ParsedRelayConnectionArguments {
  const paginationArgs: ParsedRelayConnectionArguments = {
    cursor: null,
    direction: "FORWARD",
    limit: 0,
  };

  if (isNotNullish(args.first)) {
    if (isNotNullish(args.last)) {
      throw new GraphQLError(
        "Argument first cannot be provided with argument last.",
      );
    }
    if (isNotNullish(args.before)) {
      throw new GraphQLError(
        "Argument before cannot be provided with argument first.",
      );
    }
    if (args.first > env.LIMIT) {
      throw new GraphQLError(
        `Argument first cannot not be greater than ${env.LIMIT}.`,
      );
    }
    paginationArgs.direction = "FORWARD";
    paginationArgs.limit = args.first;
    if (isNotNullish(args.after)) {
      paginationArgs.cursor = args.after;
    }
  } else if (isNotNullish(args.last)) {
    if (isNotNullish(args.after)) {
      throw new GraphQLError(
        "Argument after cannot be provided with argument last.",
      );
    }
    if (args.last > env.LIMIT) {
      throw new GraphQLError(
        `Argument last cannot be greater than ${env.LIMIT}.`,
      );
    }
    paginationArgs.direction = "BACKWARD";
    paginationArgs.limit = args.last;
    if (isNotNullish(args.before)) {
      paginationArgs.cursor = args.before;
    }
  } else {
    throw new GraphQLError(
      `Either argument first or argument last must be provided.`,
    );
  }

  return paginationArgs;
}

```

Reply:
Respected @xoldyckk sir Thank you for providing me code logic ,I want to ask that do I need to use this validation in my query code as I haven't used "Forward"
and "Backward" in my code logic ,do I need to change my code logic in the quey or just paste this code in validation file for future use ?

Reply:
read this:- https://relay.dev/graphql/connections.htm#sec-Pagination-algorithm

take example of messages in a slack channel

on slack when loading older messages scrolling upwards, that is backward pagination

on slack when loading newer messages scrolling downwards, that is forward pagination((though in chat apps this is achieved through ws/sse and not the client manually fetching a list of latest messages)

though this forward/backward naming wouldn't make sense if you provided the slack clients ability to sort the messages in ascending/descending order

Reply:
Thank you sir for providing me resources , although I have understood the logic of forward and backward pagination I just wanted to ask do I need to change my code logic according to the validation logic which you have send me?

Reply:
Respected @xoldyckk  sir what would be the value of LIMIT variable ?